### DISCUSSION TITLE
Welcome to lomap_en3b Discussions about the apllication look!

### DISCUSSION THREAD

**Discussion by RaduSeba** [Arquisoft/lomap_en3b#44] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to discuss between us the project. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

 Vamos!

**Comment by RaduSeba** [comment]
Hola. I started a discussion. I don't know how it works but I hope it is helpful.

**Comment by RaduSeba** [comment]
# Me and batu discussed the new app design.
![picture](https://user-images.githubusercontent.com/98213239/226015368-e0566cdd-53bb-44f1-89ef-f9664b780a98.jpg)
![WhatsApp Image 2023-03-17 at 20 40 25](https://user-images.githubusercontent.com/98213239/226015587-a60ff084-3173-468e-8618-3cd5be4ac25f.jpg)

## Changes made 
- First when the app starts we will load the map with our private markers and the user will be able to see in the menu bar the options Home/Friends/Map and the add marker button and logout. In this stage the user cannot interact with the map.
- Then we said that when the user presses on of the options the look of the app will change as in the right part we will have loaded a new component that can be for example:
                   - The add button will open form that will contain the fields from the second picture. We need to talk about the categories we want and if we change the fields or the look
                   - The friends menu will let you see your fiendlist and add new ones
                   - The home menu will let you see your profile.
                   - The map options hides the rest of the pop up menus you have
      
- After the user presses the add button he will be able to click on the map to add the location and then fill the form with the necesary data that we will later store in the pod. After the operation is done he needs to click again to add a new one.
- And we said that when a user clicks on a marker a menu will pop on the right showing him the location and respective comments of the location from his friends or his.
 ### We can discuss if everydody agrees with the design so we can move forward with the development of the app.

**Comment by juanmglzs** [comment]
I do agree with everything but I have a question. Didn't we say that the friends management will be done by the user outside LoMap? 
The only things related to friends that we app will offer were setting if a new location (place) will be private or friend. 
Or is it about adding the friends' pod id? So it will be only one field asking the friends' webid. [This might be useful when talking about sharing files between users](https://forum.solidproject.org/t/we-are-not-sure-how-we-could-share-files-between-users/2780).

**Reply by RaduSeba** [reply]
I think only with the webid should be easier. Also we should be able to see our friend list.

**Reply by FdezAriasSara** [reply]
yeah I agree with everything aesthetic-wise too!  Super nice thank you :) @Batuhanbyr @RaduSeba 

 @juanmglzs , when we said that i think it was more related to 

_Are lomap's friends equal to pod friends?_

  Or 

_Can users be friends within the app but not in the pod (and vice versa)?_


We thought that the best choice was that if you are friends with someone on your pod , you are as well friends in LoMap, since we dont have a db. 

But I think usability-wise , is nice that the user can sort of manage his friend list from lomap too.



I have a question. 

Could we have like a view for friend requests? To accept them and reject them?



This is my proposal for the friends view:
![image](https://user-images.githubusercontent.com/79209608/226097260-0c153a3b-d809-4753-be29-92f65da3a45a.png)

**Reply by RaduSeba** [reply]
I think the friends should be the one's in your pod. That should be the list that we show.

And also I like the idea with friend's requests and the look.. If we are able to do that would be nice.

**Comment by FdezAriasSara** [comment]
@juanmglzs @RaduSeba  BTW we need to agree on the location schema:

 i suggest that locations have: (based on https://schema.org/Place)
name

- longitude 
- latitude
- photo(optional)
- review(optional)
- aggregateRating(optional) 
- 
**doubt: will places have coments from other users?(that should be included in the user interface too then**

and , in order to give places a category we could use **additionalProperty** where the values could be:
[Shop, Bar, Monument, Sight, Museum, Restaurant,..]
(_any other category you think of can be added of course._)

**Reply by RaduSeba** [reply]
I think for now we do markers seperately for each friend and the user . So when you click a marker you see what the user who pinned that posted. 


And for the location we need to add the option top choose if you want to share the marker with others or only private no ?
